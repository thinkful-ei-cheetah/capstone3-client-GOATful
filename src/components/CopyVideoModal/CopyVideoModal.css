.copy-video-modal .field-group {
  margin-bottom: 15px;
  color: black
}

.copy-video-modal .copy-wrapper {
  background: lightgray;
  padding: 8px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.copy-video-modal h3 {
  margin: 0;
  padding-bottom:5px;
}

.copy-video-modal .f-icon {
  cursor: pointer;
}

@media screen and (max-width:500px){
  .copy-video-modal{
    margin-top: 20px;
  }
}