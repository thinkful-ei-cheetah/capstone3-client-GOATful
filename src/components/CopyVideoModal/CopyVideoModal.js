import React, {Component} from 'react';
import Modal from 'react-modal';
import './CopyVideoModal.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faCopy, faWindowClose } from '@fortawesome/free-solid-svg-icons'
import {CopyToClipboard} from 'react-copy-to-clipboard';
import PreviewsApi from '../../services/previews-api';


const modalStyles = {
  content : {
    maxWidth              : '40%',
    width                 : '300px',
    top                   : '50%',
    left                  : '50%',
    right                 : 'auto',
    bottom                : 'auto',
    marginRight           : '-50%',
    transform             : 'translate(-50%, -50%)'
  }
};

// Fix for React Modal testing
if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#root');
  
export default class CopyVideoModal extends Component {
  static defaultProps = {
    video: {},
    isOpen: false,
    onRequestClose: () => {}
  }

  state = {
    activePreview: {tags: []},
    error: null
  }

  getActivePreview = async () => {
    const { video } = this.props
    try {
      const activePreview = await PreviewsApi.getActivePreview(video.id)
      activePreview.youtube_display_name = video.youtube_display_name
      activePreview.tags = video.tags
      this.setState({activePreview})
    } catch(err) {
      this.setState({error: err.message})
    }
  }

  renderCopyableFields = (activePreview) => {
    if (this.state.error) {
      return (
        <div className='error'>
          An error occurred while fetching your currently active video. <p>The server responded with: {this.state.error}</p>
        </div>
      )
    } else {
      return (
        <>
          <h2 className="add-video-header">Your Current Active Video</h2>
          <div className='copy-video-modal'>
            <div className='field-group'>
              <h3>Title</h3>
              <div className='copy-wrapper'>
                <div className='copy-field'>{activePreview.title}</div>
                <CopyToClipboard text={activePreview.title}>
                  <FontAwesomeIcon className="f-icon" icon={faCopy} />
                </CopyToClipboard>
              </div>
            </div>
            <div className='field-group'>
              <h3>Description</h3>
              <div className='copy-wrapper'>
                <div className='copy-field'>{activePreview.description}</div>
                <CopyToClipboard text={activePreview.description}>
                  <FontAwesomeIcon className="f-icon" icon={faCopy} />
                </CopyToClipboard>
              </div>
            </div>
            <div className='field-group'>
              <h3>Youtube Display Name</h3>
              <div className='copy-wrapper'>
                <div className='copy-field'>{activePreview.youtube_display_name}</div>
                <CopyToClipboard text={activePreview.youtube_display_name}>
                  <FontAwesomeIcon className="f-icon" icon={faCopy} />
                </CopyToClipboard>
              </div>
            </div>
            <div className='field-group'>
              <h3>Tags</h3>
              <div className='copy-wrapper'>
                <div className='copy-field'>{activePreview.tags.join(', ')}</div>
                <CopyToClipboard text={activePreview.tags.join(', ')}>
                  <FontAwesomeIcon className="f-icon" icon={faCopy} />
                </CopyToClipboard>
              </div>
            </div>
          </div>
        </>
      );
    }
  }

  render() {
    const { activePreview } = this.state
    
    return (
      <Modal
        isOpen={this.props.isOpen}
        onAfterOpen={this.getActivePreview}
        onRequestClose={this.props.onRequestClose}
        style={modalStyles}
        contentLabel={'Copy Video Data'}
        closeTimeoutMS={200}
      >
        {this.renderCopyableFields(activePreview)}
        <span className='close-modal-btn' onClick={this.props.onRequestClose}>
          <FontAwesomeIcon icon={faWindowClose}/>
        </span>
      </Modal>
    )
  }
}
  