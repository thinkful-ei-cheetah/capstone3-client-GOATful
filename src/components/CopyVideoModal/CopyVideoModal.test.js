import React from 'react';
import ReactDOM from 'react-dom';
import CopyVideoModal from './CopyVideoModal';
import renderer from 'react-test-renderer';

describe('CopyVideoModal Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(< CopyVideoModal />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<CopyVideoModal/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
