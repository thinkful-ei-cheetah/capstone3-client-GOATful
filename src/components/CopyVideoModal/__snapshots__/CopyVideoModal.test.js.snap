// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyVideoModal Component renders the UI as expected 1`] = `null`;
